truffle/src/main/ruby/core/rubinius/api/shims/mirror.rb

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+module Rubinius
+  class Mirror
+
+    # Implementation of the :module_mirror Rubinius primitive.
```

Comment:
I think Chris wants me to switch back to pure Java, so this may be moot.  But we would need the Java one to dispatch from the primitive manager currently.  The manager could be updated of course to avoid the extra level of indirection, but it would probably complicate things.